subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_internal.h

```diff
@@ -10,6 +10,13 @@ int lll_adv_data_release(struct lll_adv_pdu *pdu);
 
 struct pdu_adv *lll_adv_pdu_alloc(struct lll_adv_pdu *pdu, uint8_t *idx);
 
+#if IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT_PDU_EXTRA_DATA_MEMORY)
```

Comment:
Lets stop conditional compilation for declarations and only enclose definitions. Let the linker throw error when a symbol is undefined.

Reply:
#if defined removed.